test/test_public_bindings.py

```diff
@@ -256,6 +256,7 @@ def test_modules_can_be_imported(self):
             "torch.onnx._internal.fx.op_validation",
             "torch.onnx._internal.fx.passes",
             "torch.onnx._internal.fx.type_utils",
+            "torch.testing._internal.common_state_dict",
```

Comment:
Thanks for the review! It's my understanding that under mac builds, we don't have support for distributed being built by default. I was following the lead of the imports listed in the same ignore list here - https://github.com/pytorch/pytorch/blob/79e3833703c421cb9cb9ec2c45751a09abbe6e9b/test/test_public_bindings.py#L264C1-L266C74 - which actually end up being downstream deps of `common_state_dict` anyway.

Reply:
Sure, I would still argue it's best if the module can be imported, even if the functions in it raise nice errors when trying to use distributed error.

Reply:
Updating here for visibility!

We chatted offline and although the current state isn't ideal, since this is a private implementation the current state is OK. 

Additionally, I think the error really needs to be fixed in the `ShardedTensor` import since that's what's causing the import errors on Mac, which is a little out of scope for this PR.